range_tombstone.cc

```diff
@@ -60,6 +60,11 @@ position_in_partition_view range_tombstone::position() const {
     return position_in_partition_view(position_in_partition_view::range_tombstone_tag_t(), start_bound());
 }
 
+position_in_partition_view range_tombstone::position_reversed() const {
+    const bound_view bv(end, reverse_kind(end_kind));
+    return position_in_partition_view(position_in_partition_view::range_tombstone_tag_t(), bv);
```

Comment:
Why can't we return `end_position()` ?

position_in_partition is bound-agnostic, it is a point in some key domain.

Reply:
We can, I wasn't aware that the comparator is bound-agnostic.

Reply:
Are you sure about it?
The bound kind is used to set the `_bound_weight`:
```cpp
    position_in_partition_view(range_tag_t, bound_view bv)
        : _type(partition_region::clustered), _bound_weight(position_weight(bv.kind())), _ck(&bv.prefix()) { }
```
The comparator later on uses the `_bound_weight`:
```cpp
            return _cmp(*a._ck, int8_t(a._bound_weight), *b._ck, int8_t(b._bound_weight));
```

Reply:
The information in bound kind is used because it is relevant. But _bound_weight doesn't know if it's a start or end bound, it only holds information about which side relative to the key the position is. You can have this position as a start or end bound just fine.

position_in_partition extends the domain of keys with objects which are before/after keys and prefixes.

Reply:
`Why can't we return end_position() ?` -> because `end is before ck 1` inverted is `start is after ck 1`, and not `start is before ck 1`. I'm sorry, but I don't get why you ask such a question.

Reply:
Right. position_in_partition is bound-agnostic, but it's not reversal-invariant. When reversing schema, we have to reverse the position. Which makes me wonder whether we should do it for clustering_row fragments as well. Currently not, because their positions are reversal-invariant.

Still, `position_reversed()` could return `end_position().reversed()` (or we could inline this to the caller, there's no extra logic there). We're moving away from the bound_view API because it has its unique issues.

Reply:
To be safe, we should call reverse() on all positions, it would just be no-op for rows.